source/gfoidl.Base64/Internal/Base64Encoder/Base64Encoder.Encode.netcoreapp.cs

```diff
@@ -36,18 +36,172 @@ public partial class Base64Encoder
+                    avxTranslationContant0 = Vector256.Create((byte)51);
+                    avxTranslationContant1 = Vector256.Create((sbyte)25);
+                    avxLut                 = AvxEncodeLut.ReadVector256();
+
+                    srcBytes.AssertRead<Vector256<sbyte>, byte>(ref srcStart, maxSrcLength);
+                    Vector256<sbyte> str = srcBytes.ReadVector256();
+
+                    // shift by 4 bytes, as required by enc_reshuffle
+                    str      = Avx2.PermuteVar8x32(str.AsInt32(), AvxEncodePermuteVec.ReadVector256().AsInt32()).AsSByte();
+                    srcBytes = ref Unsafe.Subtract(ref srcBytes, 4);
+
+                    while (true)
+                    {
+#if DEBUG
+                        Avx2EncodingIteration?.Invoke();
+#endif
+                        // Reshuffle
+                        str                  = Avx2.Shuffle(str, avxShuffleVec);
+                        Vector256<sbyte> t0  = Avx2.And(str, avxShuffleConstant0);
+                        Vector256<sbyte> t2  = Avx2.And(str, avxShuffleConstant2);
+                        Vector256<ushort> t1 = Avx2.MultiplyHigh(t0.AsUInt16(), avxShuffleConstant1);
+                        Vector256<short> t3  = Avx2.MultiplyLow(t2.AsInt16(), avxShuffleConstant3);
+                        str                  = Avx2.Or(t1.AsSByte(), t3.AsSByte());
+
+                        // Translation
+                        Vector256<byte> indices = Avx2.SubtractSaturate(str.AsByte(), avxTranslationContant0);
+                        Vector256<sbyte> mask   = Avx2.CompareGreaterThan(str, avxTranslationContant1);
+                        Vector256<sbyte> tmp    = Avx2.Subtract(indices.AsSByte(), mask);
+                        str                     = Avx2.Add(str, Avx2.Shuffle(avxLut, tmp));
+
+                        dest.AssertWrite<Vector256<sbyte>, T>(ref destStart, destLength);
+                        dest.WriteVector256(str);
+
+                        srcBytes = ref Unsafe.Add(ref srcBytes, 24);
+                        dest     = ref Unsafe.Add(ref dest    , 32);
+
+                        if (Unsafe.IsAddressGreaterThan(ref srcBytes, ref simdSrcEnd))
+                            break;
+
+                        srcBytes.AssertRead<Vector256<sbyte>, byte>(ref srcStart, maxSrcLength);
+                        str = srcBytes.ReadVector256();
+                    }
+
+                    sourceIndex = (uint)(ulong)Unsafe.ByteOffset(ref srcStart, ref srcBytes) + 4;
+                    destIndex   = (uint)(ulong)Unsafe.ByteOffset(ref destStart, ref dest) / (uint)Unsafe.SizeOf<T>();
+                    srcBytes    = ref srcStart;
+                    dest        = ref destStart;
+#if DEBUG
+                    Avx2Encoded?.Invoke();
+#endif
                     if (sourceIndex == srcLength)
                         goto DoneExit;
                 }
+                else
+                {
+                    avxShuffleVec          = default;
```

Comment:
This will emit
```asm
G_M8745_IG09:
       C5FC57C0             vxorps   ymm0, ymm0
       C5F457C9             vxorps   ymm1, ymm1
       C5EC57D2             vxorps   ymm2, ymm2
       C5E457DB             vxorps   ymm3, ymm3
       C5DC57E4             vxorps   ymm4, ymm4
       C5D457ED             vxorps   ymm5, ymm5
       C5CC57F6             vxorps   ymm6, ymm6
       C5C457FF             vxorps   ymm7, ymm7
						;; bbWeight=0.50 PerfScore 1.33
```
Right now I don't know any way for C# to let me use these variables w/o initialization.